zerver/lib/narrow.py

```diff
@@ -1042,6 +1056,12 @@ def parse_anchor_value(anchor_val: Optional[str], use_first_unread_anchor: bool)
         return LARGER_THAN_MAX_MESSAGE_ID
     if anchor_val == "first_unread":
         return None
+    if anchor_val == "date":
+        if not anchor_date:
+            raise JsonableError(_("Missing 'anchor_date' argument."))
+        # This anchor type is used to search for the message id closes to the anchor_date.
+        return None
```

Comment:
Would it make sense to do the `not anchor_date` check inside `parse_anchor_date_value`?

I kinda feel like we should only be parsing `anchor_date` if `anchor` was set in a way such that it should be used.

Reply:
I have moved the validation logic inside `parse_anchor_date_value`.